### PR TITLE
Use FileInputStream/FileOutputStream for System.{in, out, err}

### LINKED ISSUE #646: Unexpected "reordering" of Thread.sleep and System.out calls
Running code such as the following:
```scala
object Main {
  def main(args: Array[String]): Unit = {
    println("1")
    Thread.sleep(1000)
    println("2")
    Thread.sleep(1000)
    println("3")
  }
}
```
exhibits the behavior of waiting two full seconds (producing no output), and then prints
```
1
2
3
```
at the same time.

### PR DESCRIPTION
**Pull request by densh**
We've accidentally duplicated effort on implementing the same code
twice: once for File I/O and once for the standard input/output/error
streams.

Fixes #646

### CODE REVIEW

File: javalib/src/main/scala/java/lang/System.scala [hunk 1]
```diff
@@ -132,50 +135,4 @@ object System {
...
-        nread = unistd.read(unistd.STDIN_FILENO, buf, count)
-        if (nread == -1 && errno.errno != EINTR)
-          throw new IOException("Error on reading stdin")
-      } while (nread == -1)
-      nread
-    }
-
-    def read(): CInt = {
-      val buffer = stackalloc[scala.Byte](1)
-      var nread  = readAndRetry(unistd.STDIN_FILENO, buffer, 1)
```
Comment (jonas) [inline javalib/src/main/scala/java/lang/System.scala]: I suggest to port the error checking and retry logic to FileInputStream. Right now it simply ignore errors.

Comment (densh) [inline javalib/src/main/scala/java/lang/System.scala]: Good catch! /cc @Duhemm

Comment (densh) [inline javalib/src/main/scala/java/lang/System.scala]: Improved in https://github.com/scala-native/scala-native/pull/659/commits/d8e249f0df4a29033c66f14ec13ff053b9a7602c

File: javalib/src/main/scala/java/io/FileOutputStream.scala [hunk 2]
```diff
@@ -17,26 +17,43 @@ class FileOutputStream(fd: FileDescriptor) extends OutputStream {
...
+    }
+
+    // we use the runtime knowledge of the array layout to avoid
+    // intermediate buffer, and read straight from the array memory
+    val buf        = buffer.asInstanceOf[runtime.ByteArray].at(offset)
+    val writeCount = unistd.write(fd.fd, buf, count)
+
+    if (writeCount < 0) {
+      // negative value (typically -1) indicates that write failed
+      throw new IOException("couldn't write to file")
```
Comment (jonas) [inline javalib/src/main/scala/java/io/FileOutputStream.scala]: I think it still makes sense to retry if `errno == EINTR`. Will make a PR if nobody beats me to it.

Comment (densh) [inline javalib/src/main/scala/java/io/FileOutputStream.scala]: Go for it!

**Pull request merged**